From 88e5cd319ce65e478ddfca00a7e8ac77d6067b11 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Felix=20Paul=20K=C3=BChne?= <fkuehne@videolan.org>
Date: Fri, 31 Jul 2026 15:39:05 +0200
Subject: [PATCH 21/27] playlist/podcast: expose more global metadata

This exposes link, copyright, language and author on the global feed as
metadata as required for the media library to process
---
 modules/demux/playlist/podcast.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/modules/demux/playlist/podcast.c b/modules/demux/playlist/podcast.c
index 749687e4d0..1dda8421f7 100644
--- a/modules/demux/playlist/podcast.c
+++ b/modules/demux/playlist/podcast.c
@@ -239,12 +239,20 @@ static int ReadDir( stream_t *p_demux, input_item_node_t *p_subitems )
     else if( !strcmp( psz_elname, name ) ) \
         input_item_AddInfo( p_current_input, _("Podcast Info"), \
                             info, "%s", node );
-                    ADD_GINFO( _("Podcast Link"), "link" )
-                    ADD_GINFO( _("Podcast Copyright"), "copyright" )
+#define ADD_GINFO_META( info, name, meta ) \
+    else if( !strcmp( psz_elname, name ) ) { \
+        input_item_AddInfo( p_current_input, _("Podcast Info"), \
+                            info, "%s", node ); \
+        input_item_SetMeta( p_current_input, (meta), node ); }
+                    ADD_GINFO_META( _("Podcast Link"), "link", vlc_meta_URL )
+                    ADD_GINFO_META( _("Podcast Copyright"), "copyright", vlc_meta_Copyright )
+                    ADD_GINFO_META( _("Podcast Language"), "language", vlc_meta_Language )
+                    ADD_GINFO_META( _("Podcast Author"), "itunes:author", vlc_meta_Artist )
                     ADD_GINFO( _("Podcast Category"), "itunes:category" )
                     ADD_GINFO( _("Podcast Keywords"), "itunes:keywords" )
                     ADD_GINFO( _("Podcast Subtitle"), "itunes:subtitle" )
 #undef ADD_GINFO
+#undef ADD_GINFO_META
                     else if( !strcmp( psz_elname, "itunes:summary" ) ||
                              !strcmp( psz_elname, "description" ) )
                     { /* <description> isn't standard iTunes podcast stuff */
-- 
2.50.1 (Apple Git-155)

